token-metadata/program/src/utils.rs

```diff
@@ -940,6 +933,73 @@ pub fn process_create_metadata_accounts_logic(
     Ok(())
 }
 
+pub fn check_token_standard(
+    metadata: Metadata,
+    edition_account_info: Option<&AccountInfo>,
+    program_id: &Pubkey,
+) -> Option<TokenStandard> {
+    if let Some(val) = edition_account_info {
+        let mint_decimals = get_mint_decimals(val).ok()?;
```

Comment:
`get_mint_decimals` expects the `AccountInfo` of an actual [mint account](https://docs.rs/spl-token/latest/spl_token/state/struct.Mint.html) so you can't pass in the master edition or edition account here. 

I can go into more details of what this function is doing if you want the nitty gritty details but it's basically referencing directly into the Mint account layout to get the decimals value.